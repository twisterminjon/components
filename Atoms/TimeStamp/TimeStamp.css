.timestamp {
  font-size: 12px;
  color: var(--messagepage__usercardthread_read_fg);
  padding: 5px;
  white-space: nowrap;
  overflow: hidden;
  pointer-events: none;
}

.timestamp--right {
  margin-left: auto;
}
