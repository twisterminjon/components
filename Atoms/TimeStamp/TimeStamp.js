import React, { Component } from 'react'
import PropTypes from 'prop-types'

import friendlyTime from 'friendly-time'
import dayjs from 'dayjs'

import './TimeStamp.css'

export default class TimeStamp extends Component {
  static propTypes = {
    /** A timestamp in UTC ISO format */
    timestamp: PropTypes.string.isRequired,
    /** ClassName for the wrapper */
    className: PropTypes.string,
    /** Style for the wrapper */
    style: PropTypes.object,
  }
  static defaultProps = {
    timestamp: '',
    className: '',
    style: {},
  }

  render() {
    const { timestamp, className, style, ...rest } = this.props

    // convert timestamp to local time and then friendly it up!
    const localTimeStamp = dayjs
      .utc(timestamp)
      .local()
      .toDate()
    const niceTime = friendlyTime(localTimeStamp)

    return (
      <span className={`timestamp  ${className}`} style={style} {...rest}>
        {niceTime}
      </span>
    )
  }
}
