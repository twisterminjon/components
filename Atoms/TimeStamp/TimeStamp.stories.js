import React from 'react'

import { storiesOf } from '@storybook/react'
import centered from '@storybook/addon-centered/react'
import { withKnobs, text } from '@storybook/addon-knobs'

import TimeStamp from './TimeStamp'

const stories = storiesOf('Provider/v1/Atoms', module)
stories.addDecorator(withKnobs)

stories.addDecorator(centered).add('TimeStamp', () => {
  const timestampLabel = 'timestamp'
  const timestampDefault = '2018-12-05T10:30:15'
  const timestamp = text(timestampLabel, timestampDefault)

  return (
    <div
      style={{
        width: 200,
      }}>
      <TimeStamp timestamp={timestamp} />
    </div>
  )
})
