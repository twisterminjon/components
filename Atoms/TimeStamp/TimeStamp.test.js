import React from 'react'
import renderer from 'react-test-renderer'
import TimeStamp from './TimeStamp'

describe('TimeStamp', () => {
  // We need to send a consistent timestamp to the component as
  // it will convert the timestamp into a friendly string
  // we want the same string everytime the test runs so the snapshot matches
  const now = new Date()
  now.setHours(now.getHours() - 1)
  const timestamp = now.toISOString()

  it('matches the snapshot for sent', () => {
    const wrapper = renderer.create(<TimeStamp timestamp={timestamp} />)

    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
