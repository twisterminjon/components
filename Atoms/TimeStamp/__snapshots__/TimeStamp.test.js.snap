// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeStamp matches the snapshot for sent 1`] = `
<span
  className="timestamp  "
  style={Object {}}
>
  an hour ago
</span>
`;
